emperor/support_files/js/sceneplotview3d.js

```diff
@@ -100,6 +102,90 @@ define([
     this.dimensionRanges = {'max': [], 'min': []};
     this.drawAxesWithColor(0xFFFFFF);
     this.drawAxesLabelsWithColor(0xFFFFFF);
+
+    var raycaster = new THREE.Raycaster();
+    var mouse = new THREE.Vector2();
+
+    //initialize subscribers for event callbacks
+    this.EVENTS = ['click', 'dblclick'];
+    var subscribers = {};
+    for (var i = 0; i < this.EVENTS.length; i++) {
+      subscribers[this.EVENTS[i]] = [];
+    }
+
+    //create helper function for callbacks
+    var eventCallback = function(eventType, event) {
+      event.preventDefault();
+      //dont do anything if no subscribers
+      if (subscribers[eventType].length === 0) {
+        return;
+      }
+
+      var element = scope.renderer.domElement;
+      mouse.x = ((event.clientX - element.offsetLeft) / element.width) * 2 - 1;
+      mouse.y = -((event.clientY - element.offsetTop) / element.height) * 2 + 1;
+
+      raycaster.setFromCamera(mouse, scope.camera);
```

Comment:
Ray casting will get expensive and potentially inaccurate for a drag box. What you really want to do is ask the opengl engine for what is present in your view, and resolve the "hit" object or objects from there by navigating the object hierarchy. Details are [here](http://www.glprogramming.com/red/chapter13.html) although I'm not immediately sure how to translate this to threejs